map_model/src/map.rs

```diff
@@ -1063,4 +1065,136 @@ impl Map {
         }
         result
     }
+
+    // TODO This returns a mixture of different things related to the turn. It might be better
+    // to create and return a `Turn`.
+    pub fn get_ban_turn_info(
+        &self,
+        r1: &Road,
+        r2: &Road,
+    ) -> (TurnType, Pt2D, Angle, IntersectionID) {
+        // Determine where to place the symbol
+        let i = match r1.common_endpoint(r2) {
+            CommonEndpoint::One(i) => i,
+            // This is probably rare, just pick one side arbitrarily
+            CommonEndpoint::Both => r1.src_i,
+            CommonEndpoint::None => {
+                // This may be that `r1` and `r2` are joined by a complicated_turn_restrictions,
+                // but don't have a CommonEndpoint.
+                // In this case the end of r1 appears to be the most appropriate location to pick here
+                r1.dst_i
+            }
+        };
+
+        // Determine what type of turn is it?
+        let (sign_pt, mut r1_angle) = r1
+            .center_pts
+            .must_dist_along((if r1.src_i == i { 0.2 } else { 0.8 }) * r1.center_pts.length());
+
+        // Correct the angle, based on whether the vector direction is towards or away from the intersection
+        // TODO what is the standard way of describing the vector direction (rather than the traffic direction) for roads?
+        r1_angle = if r1.src_i == i {
+            r1_angle.rotate_degs(180.0)
+        } else {
+            r1_angle
+        };
+
+        let (_, mut r2_angle) = r2
+            .center_pts
+            .must_dist_along((if r2.src_i == i { 0.2 } else { 0.8 }) * r2.center_pts.length());
+
+        // Correct the angle, based on whether the vector direction is towards or away from the intersection
```

Comment:
We do this sort of logic everywhere in osm2streets, and @BudgieInWA has some ideas (written somewhere...) for expressing better. We can look at it in the future